### DISCUSSION TITLE
SAE schaltet seit Update auf 2.0.1 Tasmota steckdose nicht mehr --> gelbe Lampe

### DISCUSSION THREAD

**Discussion by zeron850** [camueller/SmartApplianceEnabler#175] [Q&A]
Hallo Alex,

ich habe wir von dir beschrieben meine SAE Installation umgezogen. Das hat wunderbar funktioniert. Die config Files wurde selbst beim Update auf 2.0.1 verwendet. Die Daten kommen wie gewohnt im Sunnyportal an, leider schaltet aber der SAE nicht mehr von der gelben blinkenden Ampel während der Schaltzeit auf die grün blinkende, trotz vorhandenem PV-Überschuss. Muss ich in den Einstellungen noch was ändern?
![Bild4](https://user-images.githubusercontent.com/68141401/151040720-99a72316-e5b6-4904-be78-561237c326d5.JPG)
![Bild1](https://user-images.githubusercontent.com/68141401/151040725-25da800a-ddfa-4ae6-91e5-e8932efbf82d.JPG)
![Bild2](https://user-images.githubusercontent.com/68141401/151040727-c89f63c8-5425-46f9-aa99-495dbaed8262.JPG)
![Bild3](https://user-images.githubusercontent.com/68141401/151040730-051390b2-941f-4920-bdf0-5a4c8b62614e.JPG)

Vielen Dank!

**Comment by camueller** [comment]
Der SAE schaltet nicht, sondern gibt nur die Bedarfsanforderung auf Basis des Zeitplanes an den SHM weiter. Du musst also [an der /semp-Schnittstelle prüfen](https://github.com/camueller/SmartApplianceEnabler/blob/master/doc/SEMP_DE.md), ob dort ein Bedarf gemeldet wird. Wenn dem so ist, kann man bei vorhandenem Überschuss mit einem Einschaltbefehl vom SHM rechnen. Erst wenn dieser kommt und das Geräte eingeschaltet wird, kommt es zum grünen Blinken, wenn dabei Überschussenergie verwendet wird.

**Reply by zeron850** [reply]
Ok, danke. Ich habe dir mal die SEMP.xml beigefügt. Konnte da nichts auffälliges finden. Allerdings wird der SAE im Sunnyportal nicht mehr im Anlagenlogbuch gezeigt. Normalerweise kommt doch einmal pro Tag eine Meldung das dieser samt den Verbrauchern gefunden wurde
[semp.zip](https://github.com/camueller/SmartApplianceEnabler/files/7936732/semp.zip)
?

Die Verbindung muss aber klappen da die Verbräuche schön brav vom Sunnyportal erfasst werden...
![Bild](https://user-images.githubusercontent.com/68141401/151049935-a211f937-1ac4-47bb-adf9-a02d55fcacc5.jpg)

**Reply by zeron850** [reply]
Ich kann zum Beispiel auch keine neuen Geräte in den SAE aufnehmen. Er bleibt dann hängen und macht nicht mehr weiter..
Ich habe auch downgrade zu Version 1.6.18 probiert. Dort kommt das gleiche Fehlerbild

![Bild](https://user-images.githubusercontent.com/68141401/151052251-733746c0-0cf0-4d23-a85d-c38bee39c61b.jpg)
.

**Reply by zeron850** [reply]
Ich denke so langsam den Fehler eingrenzen zu können.

Der SAE speichert keine Änderungen der Konfiguration mehr.
![Bild](https://user-images.githubusercontent.com/68141401/151055946-ee03c4ff-4190-473f-b03c-ec9f88497601.jpg)
 
Kannst du dir das bitte mal ansehen?

**Comment by camueller** [comment]
Welche Änderung genau hast Du in der Web-Oberfläche vorgenommen, die nicht gespeichert wurde? Oder können überhaupt keine Änderungen gespeichert werden? Evtl. liegt es auch an den Zugriffsrechten auf die XML-Dateien!?

**Reply by zeron850** [reply]
Ich habe versucht einen neuen Verbraucher anzulegen und einen bestehenden abzuändern. Bei beiden kommt das gleiche Fehlerbild. 

Bezüglich der Schreibrechte habe ich das Verzeichnis mal mit 777 Rechten für alle User zum Test freigegeben. Das Ergebnis ist aber das gleiche… 🤔

**Reply by camueller** [reply]
Es muss aber etwas Docker-spezifisches sein, weil ich auch für die Version 2.0 automatisierte Tests laufen lasse, die ohne eine Konfiguration starten (also keine XML-Dateien vorhanden) und dann Geräte, Zähler, Schalter und Zeitpläne für die diversen Ausprägungen anlegen. Bei jedem Test werden Daten in die Felder gefüllt und gespeichert, dann auf die Status-Seite gewechselt, dann wieder auf die zuvor gefüllte Seite zurückgewechselt und alle Feldinhalte mit den zuvor eingefüllten Feldinhalten verglichen.

**Reply by zeron850** [reply]
Ich nutze aber gar kein Docker. Ich habe auf dem neuen Pi die manuelle Installationsroutine (wie beim alten Pi) verwendet. Könnte es u.a.  an der Java Version liegen? Ich nutze das Debianderivat Dietpi in der Version 8.0 Bullseye. Habe ich aber davor auch verwendet…

**Reply by camueller** [reply]
Ich nutze den SAE 2.0 auf Raspbian BullsEye, daran kann es also nicht liegen. Java-Version ist weiterhin Java 11, welches im Raspbian-Repo enthalten ist. Du nutzt offensichtlich nicht Raspbian. Möglilcherweise liegt darin das Problem, weil die von Dir genannten Probleme bei mir nicht auftreten.

**Reply by zeron850** [reply]
Ok, ich werde mal versuchen ein anderes Derivat zu verwenden und prüfen ob dann das Problem erneut auftritt. Danke aber für deine Hilfe!

**Reply by camueller** [reply]
Hast Du eigentlich mal die Logs auf Fehlermeldungen geprüft? Wenn das Speichern bei Dir nicht klappt, würde ich vermuten, dass sich was im Log findet. 
Ansonsten: Warum verwendest Du nicht einfach Raspbian? Das ist der offizielle Software-Stack.

**Reply by zeron850** [reply]
Ich habe mir die Logs mal angesehen und dir auch beigefügt. Es gibt nur eine Warn-Meldung. 

Der Grund für DietPi ist das ich umsteigen möchte auf einen O
[rolling-2022-01-26.log](https://github.com/camueller/SmartApplianceEnabler/files/7943840/rolling-2022-01-26.log)
droid und weg von einem Pi. Das funktioniert soweit auch gut und der SAE läuft dort nur das speichern und schalten eben nicht.

**Reply by zeron850** [reply]
Einen Fehler konnte ich abstellen. Es lag an entzogenen Schreibrechten. Das speichern funktioniert nun wieder.
Jetzt noch die Schaltthematik aus dem Sunnyportal und ich bin happy :-)

**Comment by camueller** [comment]
In Deinem /semp-XML finden sich zwei Timeframes:
```
    <PlanningRequest>
        <Timeframe>
            <DeviceId>F-20210924-000000000001-00</DeviceId>
            <EarliestStart>33212</EarliestStart>
            <LatestEnd>78212</LatestEnd>
            <MinRunningTime>0</MinRunningTime>
            <MaxRunningTime>45000</MaxRunningTime>
        </Timeframe>
        <Timeframe>
            <DeviceId>F-20210924-000000000001-00</DeviceId>
            <EarliestStart>119612</EarliestStart>
            <LatestEnd>164612</LatestEnd>
            <MinRunningTime>0</MinRunningTime>
            <MaxRunningTime>45000</MaxRunningTime>
        </Timeframe>
    </PlanningRequest>
```
Wie Du sehen kannst, beginnt der erste *frühestens* 33212 Sekunden, nachdem Du gestern das /semp-XML gespeichert hast.
Der SHM wird Dir also frühestens nach dieser Zeit einen Einschalt-Befehl schicken. Du solltest mal Deine Zeitpläne überprüfen und/oder  Datum und Zeit auf Deinem Raspi.

**Reply by zeron850** [reply]
Danke dir für die Info. Ich habe die Zeit im Pi gecheckt. Die zeigt genau die Systemzeit meines Notebooks also ist der Punkt mal sauber.
Die Einschaltzeiten habe ich wie folgt definiert:
![grafik](https://user-images.githubusercontent.com/68141401/151220113-0694912a-f857-4420-b942-01cab2c0a7f0.png)

**Reply by zeron850** [reply]
Sind da wie ich sehen kann zwei Zeitpläne gespeichert, obwohl nur einer im SAE angezeigt wird?

**Reply by camueller** [reply]
Der SAE meldet dem SHM die Anforderungen der nächsten 48 Stunden. Das kannst Du auch an den Werte des 2. Timeframes sehen, dass die einen Tag später sind. Entscheidend ist, dass Du nur einen Einschalt-Befehl vom SHM erwarten kannst, wenn für einen Timeframe EarliestStart=0 ist. Das kannst Du alles auch unter [Fragen und Antworten](https://github.com/camueller/SmartApplianceEnabler/blob/master/doc/QA_DE.md#semp3) nachlesen und hat nichts mit den Änderungen der SAE-Version 2.0 zu tun.

**Reply by zeron850** [reply]
Ok, danke dir!
